### ISSUE TITLE
make ios-deploy optional?

### ISSUE THREAD

**Issue by ruisalgado** [kabisa/maji#282]
building `ios-deploy` is riddled with issues in MacOS Catalina, from requiring `--unsafe-perm=true` to issues linking to xcode libraries during the build step. Its maintainers now recommend installing a pre-compiled binary via `brew`.
The current maji setup has the installation of `ios-deploy` and `ios-sim` 
be triggered by a cordova hook, which means they get installed even if you're never going to use them (you wouldn't need them for example in a CI environment).
A better (?) approach could be to try to resolve these dependencies only when they are need (when running certain `maji` cli commands). At this point we might offer to install `ios-deploy` via brew as an alternative to npm, or at least educate the user about it.

**Comment by pascalw** [comment]
Yeah `ios-deploy` has always been a PITA, but I didn't realize it had gotten even worse on Catalina.

I'm in favor of removing the code that installs the module automatically and just putting it as a prerequisite in the docs. WDYT?

**Comment by ruisalgado** [comment]
sounds good to me
